# Exploration of Genomic mutation Dataset

Here, I perform initial exploration of the mutation dataset.
Inspect
* structure
* column meanings
* missing data
* statistics



In [50]:
# Standard libraries
import pandas as pd
import numpy as np

# Visualisation
import matplotlib.pyplot as plt
import seaborn as sns

# Display settings
# Display more rows and columns
pd.set_option("display.max_rows", 400)
pd.set_option("display.max_columns", 400)
pd.set_option("display.width", 1000)

# Optional: Plot styling
sns.set(style="whitegrid")

In [51]:
# Path to your data
file_path = "../data/raw/data_mutations.txt"

# Read the file
df = pd.read_csv(file_path, sep="\t", comment="#", low_memory=False)

In [52]:
# Preview
df.head(10)

Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  End_Position Strand           Consequence Variant_Classification Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2      dbSNP_RS  dbSNP_Val_Status Tumor_Sample_Barcode Matched_Norm_Sample_Barcode  Match_Norm_Seq_Allele1  Match_Norm_Seq_Allele2  Tumor_Validation_Allele1  Tumor_Validation_Allele2  Match_Norm_Validation_Allele1  Match_Norm_Validation_Allele2  Verification_Status  Validation_Status Mutation_Status  Sequencing_Phase  Sequence_Source  Validation_Method  Score  BAM_File  Sequencer  t_ref_count  t_alt_count  n_ref_count  n_alt_count                              HGVSc               HGVSp    HGVSp_Short    Transcript_ID          RefSeq Protein_position       Codons Exon_Number  1000G_AF  1000G_AFR_AF  1000G_AMR_AF  1000G_EAS_AF  1000G_EUR_AF  1000G_SAS_AF APPRIS Allele Amino_acids         BIOTYPE CANONICAL         CCDS    CDS_position                                           CLIN_SIG  \
0       MTCL1           23255     BI     GRCh38         18         8806948       8806948      +      missense_variant      Missense_Mutation          SNP                G                 G                 A   rs199934266               NaN     TCGA-AB-2881-03A            TCGA-AB-2881-11A                     NaN                     NaN                       NaN                       NaN                            NaN                            NaN                  NaN                NaN         Somatic               NaN              NaN                NaN    NaN       NaN        NaN           26            6          NaN          NaN       ENST00000306329.15:c.3449G>A        p.Arg1150His       p.R1150H  ENST00000306329  NM_001378206.1             1150      cGt/cAt        9/14       NaN           NaN           NaN           NaN           NaN           NaN     P3      A         R/H  protein_coding       YES          NaN       3449/5718                                                NaN   
1       HSH2D           84941     BI     GRCh38         19        16157257      16157257      +     inframe_insertion           In_Frame_Ins          INS                G                 G              CTCC         novel               NaN     TCGA-AB-2881-03A            TCGA-AB-2881-11A                     NaN                     NaN                       NaN                       NaN                            NaN                            NaN                  NaN                NaN         Somatic               NaN              NaN                NaN    NaN       NaN        NaN           89            9          NaN          NaN  ENST00000613986.4:c.522delinsCTCC         p.Ser175dup      p.S175dup  ENST00000613986  NM_001352265.2              175   ccG/ccCTCC         6/6       NaN           NaN           NaN           NaN           NaN           NaN     P1   CTCC        P/PS  protein_coding       NaN  CCDS74304.1        522/1059                                                NaN   
2      PRDM16           63976     BI     GRCh38          1         3411959       3411959      +      missense_variant      Missense_Mutation          SNP                G                 G                 A   rs768733228               NaN     TCGA-AB-2895-03A            TCGA-AB-2895-11A                     NaN                     NaN                       NaN                       NaN                            NaN                            NaN                  NaN                NaN         Somatic               NaN              NaN                NaN    NaN       NaN        NaN           25           35          NaN          NaN       ENST00000270722.10:c.1762G>A         p.Val588Met        p.V588M  ENST00000270722     NM_022114.4              588      Gtg/Atg        9/17       NaN           NaN           NaN           NaN           NaN           NaN     P4      A         V/M  protein_coding       YES  CCDS41236.2       1762/3831                             uncertain_significance   
3 

In [53]:
print("shape", df.shape)
print("\nColumns:")
print(df.columns.tolist())

shape (569, 140)

Columns:
['Hugo_Symbol', 'Entrez_Gene_Id', 'Center', 'NCBI_Build', 'Chromosome', 'Start_Position', 'End_Position', 'Strand', 'Consequence', 'Variant_Classification', 'Variant_Type', 'Reference_Allele', 'Tumor_Seq_Allele1', 'Tumor_Seq_Allele2', 'dbSNP_RS', 'dbSNP_Val_Status', 'Tumor_Sample_Barcode', 'Matched_Norm_Sample_Barcode', 'Match_Norm_Seq_Allele1', 'Match_Norm_Seq_Allele2', 'Tumor_Validation_Allele1', 'Tumor_Validation_Allele2', 'Match_Norm_Validation_Allele1', 'Match_Norm_Validation_Allele2', 'Verification_Status', 'Validation_Status', 'Mutation_Status', 'Sequencing_Phase', 'Sequence_Source', 'Validation_Method', 'Score', 'BAM_File', 'Sequencer', 't_ref_count', 't_alt_count', 'n_ref_count', 'n_alt_count', 'HGVSc', 'HGVSp', 'HGVSp_Short', 'Transcript_ID', 'RefSeq', 'Protein_position', 'Codons', 'Exon_Number', '1000G_AF', '1000G_AFR_AF', '1000G_AMR_AF', '1000G_EAS_AF', '1000G_EUR_AF', '1000G_SAS_AF', 'APPRIS', 'Allele', 'Amino_acids', 'BIOTYPE', 'CANONICAL', 'CCD

In [54]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 569 entries, 0 to 568
Columns: 140 entries, Hugo_Symbol to Annotation_Status
dtypes: float64(62), int64(9), object(69)
memory usage: 622.5+ KB


In [55]:
missing_values = df.isna().mean().sort_values(ascending=False)
missing_values.head(20)

genomic_location_explanation     1.0
TRANSCRIPTION_FACTORS            1.0
n_ref_count                      1.0
Sequencer                        1.0
BAM_File                         1.0
Score                            1.0
Validation_Method                1.0
Sequence_Source                  1.0
Sequencing_Phase                 1.0
Validation_Status                1.0
Verification_Status              1.0
Match_Norm_Validation_Allele2    1.0
Match_Norm_Validation_Allele1    1.0
Tumor_Validation_Allele2         1.0
Tumor_Validation_Allele1         1.0
Match_Norm_Seq_Allele2           1.0
Match_Norm_Seq_Allele1           1.0
MOTIF_NAME                       1.0
MOTIF_POS                        1.0
RNA_ref_count                    1.0
dtype: float64

In [56]:
df.describe().T

count          mean           std            min           25%           50%           75%           max
Entrez_Gene_Id                 569.0  6.051419e+05  7.505472e+06      43.000000  4.869000e+03  2.299900e+04  8.000500e+04  1.097291e+08
Start_Position                 569.0  7.621491e+07  5.598350e+07  900111.000000  3.287844e+07  6.173562e+07  1.168521e+08  2.485930e+08
End_Position                   569.0  7.621491e+07  5.598350e+07  900111.000000  3.287844e+07  6.173562e+07  1.168521e+08  2.485930e+08
dbSNP_Val_Status                 0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Match_Norm_Seq_Allele1           0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Match_Norm_Seq_Allele2           0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Tumor_Validation_Allele1         0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Tumor_Validation_Allele2         0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Match_Norm_Validation_Allele1    0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Match_Norm_Validation_Allele2    0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Verification_Status              0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Validation_Status                0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Sequencing_Phase                 0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Sequence_Source                  0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Validation_Method                0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Score                            0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
BAM_File                         0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
Sequencer                        0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
t_ref_count                    569.0  1.041388e+02  1.273066e+02       0.000000  3.300000e+01  6.300000e+01  1.280000e+02  1.048000e+03
t_alt_count                    569.0  4.217399e+01  1.294032e+02       2.000000  1.000000e+01  2.200000e+01  4.500000e+01  2.890000e+03
n_ref_count                      0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
n_alt_count                      0.0           NaN           NaN            NaN           NaN           NaN           NaN           NaN
1000G_AF                        23.0  2.521739e-04  1.503619e-04       0.000200  2.000000e-04  2.000000e-04  2.000000e-04  8.000000e-04
1000G_AFR_AF                    23.0  5.739130e-04  8.001235e-04       0.000000  0.000000e+00  0.000000e+00  8.000000e-04  3.000000e-03
1000G_AMR_AF                    23.0  2.434783e-04  5.425747e-04       0.000000  0.000000e+00  0.000000e+00  0.000000e+00  1.400000e-03
1000G_EAS_AF                    23.0  3.043478e-04  4.704720e-04       0.000000  0.000000e+00  0.000000e+00  1.000000e-03  1.000000e-03
1000G_EUR_AF                    23.0  8.695652e-05  2.881041e-04       0.000000  0.000000e+00  0.000000e+00  0.000000e+00  1.000000e-03
1000G_SAS_AF                    23.0  1.304348e-04  3.443502e-04       0.000000  0.000000e+00  0.000000e+00  0.000000e+00  1.000000e-03
DISTANCE                         2.0  2.843000e+03  1.739483e+02    2720.000000  2.7815

In [69]:
unknown_columns = ['HGVSp_Short', 'Allele', 'One_Consequence', 'PolyPhen', 'SIFT', 'SOMATIC', 'VARIANT_CLASS']
for col in unknown_columns:
    if col in df.columns:
        print(f"\nColumn: {col}")
        print(df[col].head(5))


Column: HGVSp_Short
0     p.R1150H
1    p.S175dup
2      p.V588M
3       p.S72P
4      p.R882H
Name: HGVSp_Short, dtype: object

Column: Allele
0       A
1    CTCC
2       A
3       G
4       T
Name: Allele, dtype: object

Column: One_Consequence
0     missense_variant
1    inframe_insertion
2     missense_variant
3     missense_variant
4     missense_variant
Name: One_Consequence, dtype: object

Column: PolyPhen
0    benign(0.045)
1              NaN
2    benign(0.051)
3    benign(0.006)
4    benign(0.067)
Name: PolyPhen, dtype: object

Column: SIFT
0    tolerated(0.16)
1                NaN
2    tolerated(0.23)
3    tolerated(0.11)
4     deleterious(0)
Name: SIFT, dtype: object

Column: SOMATIC
0        NaN
1        NaN
2        0;1
3        0;1
4    0;1;1;1
Name: SOMATIC, dtype: object

Column: VARIANT_CLASS
0      SNV
1    indel
2      SNV
3      SNV
4      SNV
Name: VARIANT_CLASS, dtype: object


In [66]:
# Figure out alleles and their corresponding gene
gene = df['Hugo_Symbol']
allele = df['Allele']
gene_and_allele = df[['Hugo_Symbol','Allele']]
print(gene_and_allele.head(10))

  Hugo_Symbol Allele
0       MTCL1      A
1       HSH2D   CTCC
2      PRDM16      A
3       DCAF8      G
4      DNMT3A      T
5      CLASP2      T
6      SH3TC2      T
7      ATP10B      C
8       DOCK2      A
9        NPM1   TCTG


## Interesting 
* HGVSp_Short: 
* Allele
* One_Consequence
* PolyPhen
* SIFT
* SOMATIC
* VARIANT_CLASS
* VARIANT_CLASSIFICATION

### Key Columns Used in This Analysis



- **Hugo_Symbol:** which gene is affected by a mutation.
  The official gene name (from the HUGO Gene Nomenclature Committee).  
  *Example*: `TP53`, `NPM1`, `DNMT3A`.  
  

- **Variant_Classification:**  what kind of mutation is it.
  The type of mutation in terms of its biological effect.  
  *Examples*:  
  - `Missense_Mutation` → amino acid change  
  - `Silent` → no change to the protein  
  - `Nonsense_Mutation` → introduces a stop codon  
  - `Frame_Shift_Ins/Del` → disrupts protein reading frame  
  - `Splice_Site` → alters RNA splicing  
  

- **Tumor_Sample_Barcode:**  mutation to an individual patient and the mutation burden.
  A unique identifier for each tumor sample (patient).  
  *Example*: `TCGA-AB-2935-03A`.  
  

- **t_ref_count**  
  Number of sequencing reads supporting the **reference (normal)** allele in the tumor sample.  
  *Example*: `63` means 63 reads match the reference base.

- **t_alt_count**  
  Number of sequencing reads supporting the **alternate (mutated)** allele in the tumor sample.  
  *Example*: `45` means 45 reads match the mutation.  


### Summary of mutation

- Total rows: 569
- Unique tumor samples: 72

**Top mutated genes:**
- NPM1, TP53, DNMT3A, IDH2, and FLT3 appear most frequently, consistent with known cancer drivers.

**Mutation classes:**
- Missense mutations dominate (365), followed by silent mutations (110).
- Frameshift and nonsense mutations occur less frequently but often have higher impact.

**Mutations per sample:**
- Some patients have >30 mutations, others fewer than 10.
- This suggests heterogeneity in mutational burden.




In [79]:
key_columns = ["Hugo_Symbol", "Variant_Classification",  "t_ref_count", "t_alt_count", 'hotspot']
df_key = df[key_columns]
print(df_key.head(10))

  Hugo_Symbol Variant_Classification  t_ref_count  t_alt_count  hotspot
0       MTCL1      Missense_Mutation           26            6        0
1       HSH2D           In_Frame_Ins           89            9        0
2      PRDM16      Missense_Mutation           25           35        0
3       DCAF8      Missense_Mutation           44           35        0
4      DNMT3A      Missense_Mutation           63           45        0
5      CLASP2        Frame_Shift_Ins          144           90        0
6      SH3TC2      Missense_Mutation          146           38        0
7      ATP10B      Missense_Mutation           52           44        0
8       DOCK2            Splice_Site           66           66        0
9        NPM1        Frame_Shift_Ins           48           29        0


In [77]:
# Explore mutation-related columns
if 'Variant_Classification' in df.columns:
    print("Top mutation classifications:")
    print(df['Variant_Classification'].value_counts().head(10))

if 'Hugo_Symbol' in df.columns:
    print("\nTop mutated genes:")
    print(df['Hugo_Symbol'].value_counts().head(10))


Top mutation classifications:
Variant_Classification
Missense_Mutation    365
Silent               110
Frame_Shift_Ins       30
Nonsense_Mutation     22
Splice_Site           13
Frame_Shift_Del        9
Splice_Region          6
In_Frame_Ins           4
Intron                 4
5'Flank                2
Name: count, dtype: int64

Top mutated genes:
Hugo_Symbol
NPM1      8
TP53      7
DNMT3A    6
IDH2      5
RUNX1     4
TTN       3
GOLGA4    3
FLT3      3
FREM2     3
KRAS      3
Name: count, dtype: int64
